### PR TITLE
Fix AppOption types

### PR DESCRIPTION
**Pull request by slimbuck**
This PR updates AppOption types to class constructors (instead of class instances), which appears to appease typescript.

VSCode shows errors on these now, not sure how to fix that.

### CODE REVIEW

**Comment by mvaligursky** [comment]
I tried few variations of these (not this one yet), and something always does not work.

File: src/framework/app-options.js [hunk 1]
```diff
@@ -71,49 +71,49 @@ class AppOptions {
...
-     * @type {SoundManager}
+     * @type {SoundManager.constructor}
      */
     soundManager;
 
     /**
      * The graphics device.
      *
-     * @type {GraphicsDevice}
+     * @type {GraphicsDevice.constructor}
```
Comment (mvaligursky) [inline src/framework/app-options.js:81]: this one we pass an instance, and possibly others, not all we pass as types.

Comment (slimbuck) [reply src/framework/app-options.js]: ahh yes of course, updated.

**Comment by slimbuck** [comment]
With the latest update playcanvas.d.ts looks as follows:
```javascript
    lightmapper: any;
    /**
     * The BatchManager.
     *
     * @type {typeof BatchManager}
     */
    batchManager: any;
    /**
     * The XrManager.
     *
     * @type {typeof XrManager}
     */
    xr: any;
    /**
     * The component systems the app requires.
     *
     * @type {(typeof ComponentSystem)[]}
     */
    componentSystems: (typeof ComponentSystem)[];
    /**
     * The resource handlers the app requires.
     *
     * @type {(typeof ResourceHandler)[]}
     */
    resourceHandlers: (typeof ResourceHandler)[];

```

**Comment by slimbuck** [comment]
resource handler is being recast as an interface in types-fixup.js:
```javascript
dts = dts.replace('export class ResourceHandler', 'export interface ResourceHandler');
```

@willeastcott any idea why this is done?

**Comment by mvaligursky** [comment]
also, try to build API docs from jsdocs locally, I think I had issues with this in the past as well.

**Comment by willeastcott** [comment]
Yep, I have a feeling the docs build process fails on those typeof keywords, IIRC.

**Comment by slimbuck** [comment]
The latest version (with constructor syntax) seems to work fine when compiling typescript and also generates docs.

CI fails though.

**Comment by mvaligursky** [comment]
related https://github.com/playcanvas/engine/pull/4324

**Pull request closed without merging**